packages/expo-firebase-polyfill/src/Crypto.ts

```diff
@@ -0,0 +1,35 @@
+type TypedArray =
+  | Int8Array
+  | Uint8Array
+  | Int16Array
+  | Uint16Array
+  | Int32Array
+  | Uint32Array
+  | Uint8ClampedArray;
+
+/**
+ * A simple Crypto.getRandomValues implementation using Math.random().
+ * https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
+ * @param typedArray
+ */
+export function getRandomValues<T extends TypedArray>(typedArray: T): T {
+  if (
+    !(
+      typedArray instanceof Int8Array ||
+      typedArray instanceof Uint8Array ||
+      typedArray instanceof Int16Array ||
+      typedArray instanceof Uint16Array ||
+      typedArray instanceof Int32Array ||
+      typedArray instanceof Uint32Array ||
+      typedArray instanceof Uint8ClampedArray
+    )
+  ) {
+    throw new Error('Expected an integer typed-array');
+  }
+
+  const writeArray = new Uint8Array(typedArray.buffer);
```

Comment:
We should either not do this at all, use expo-random, or warn users that they've just generated an insecure random value. I can see someone setting up a firebase/payments app then having this side effect replace a secure side-effect, then cause a major vunrability.

Reply:
Yeah I agree, this might be an attack vector for someone have dubious intent. I did consider that but felt the risk tolerable because the Firebase JS SDK pretty much did the same thing in versions prior to `7.9.0`:
https://github.com/firebase/firebase-js-sdk/commit/a85f81dbbb7fa9b501d927f5f46e5295d4550dd5#diff-e83f92018dc3b6c3775739f5946730a1

But there might be other code that becomes dependent on this polyfill and that may also be compromised because of this.

That being said, I don't really have a good solution for this though. The `expo-random` package seems like a great candidate for proving a solid cryptographic random array, but unfortunately the API is async, and I don't see how we could use that as a polyfill.

I could imagine moving this to `expo-crypto` and then being able to load crypto as a polyfill. And then potentially also throwing a warning that `getRandomValues` is non secure.

What do you reckon?

Reply:
I'd rather we upstream the method to be used when needed when no other method is available https://github.com/firebase/firebase-js-sdk/blob/2348a259ca0e63e451f677909e71388fdb802e70/packages/installations/src/helpers/generate-fid.ts#L34 rather than creating a global side-effect and potentially overwriting some other `globa.crypto. getRandomValues()` method. 

If this doesn't seem possible then we should log a warning with a link to an https://github.com/expo/fyi post explaining why this had to be done.

Reply:
Let's keep this in the back of our mind. The crypto dependency was reverted in the Firebase JS SDK, so this is currently no longer a problem